PowerValidatedSolutions.psm1

```diff
@@ -29653,52 +29653,93 @@ Function Request-CSPToken {
 }
 Export-ModuleMember -Function Request-CSPToken
 
-Function Get-CSPPoxyAppliance {
+Function Get-CloudProxy {
+    <#
+        .SYNOPSIS
+        Request Cloud Proxies from VMware Cloud Service
+
+        .DESCRIPTION
+        The Get-CloudProxy cmdlet connects to the VMware Cloud Service and either downloads the OVA or
+        supplies the OVA link.
+
+        .EXAMPLE
+        Get-CloudProxy -environment production -type 'Cloud Proxy' -download -path <string>
+        This example shows how to download the Cloud Proxy OVA to the path provided.
+
+        .EXAMPLE
+        Get-CloudProxy -environment production -type 'Cloud Extensibility Proxy' -download -path <string>
+        This example shows how to download the Cloud Extensibility Proxy OVA to the path provided.
+
+        .EXAMPLE
+        Get-CloudProxy -environment production -type 'Cloud Proxy' -ovaUrl
+        This example shows how to obtain the URL to the Cloud Proxy OVA.
+    #>
+
     Param (
-        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$path
+        [Parameter (Mandatory = $true)] [ValidateSet("production","staging")] [ValidateNotNullOrEmpty()] [String]$environment,
+        [Parameter (Mandatory = $true)] [ValidateSet("Cloud Proxy","Cloud Extensibility Proxy")] [ValidateNotNullOrEmpty()] [String]$type,
+        [Parameter (Mandatory = $false, ParameterSetName = 'download')] [ValidateNotNullOrEmpty()] [Switch]$download,
+        [Parameter (Mandatory = $false, ParameterSetName = 'download')] [ValidateNotNullOrEmpty()] [String]$path,
+        [Parameter (Mandatory = $false, ParameterSetName = 'ovaUrl')] [ValidateNotNullOrEmpty()] [Switch]$ovaUrl
     )
 
     Try {
-
-        $uri = "https://api.mgmt.cloud.vmware.com/iaas/api/data-collectors"
-        $response = Invoke-RestMethod -Method POST -Uri $Uri -Headers $cspHeaders
-        
-        $cloudProxyUrl = $response.ovaLink
-        if ($PsBoundParameters.ContainsKey("path")) {
-            $cloudProxyOva = $path + "\" + ($response.ovaLink -Split (".com/"))[1]
+        if ($environment -eq "staging") {
+            $baseUrl = "https://api.staging.symphony-dev.com"
+        } elseif ($environment -eq "production") {
+            $baseUrl = "https://api.mgmt.cloud.vmware.com"
```

Comment:
vRealize Automation Cloud is available in multiple global regions. For organizations located outside of the United States, you must prefix the URL with the country abbreviation for your API endpoint as shown in the following examples.

Should we consider adding a validate set with a -geo and then amend the prefix to the base URI?

- United States: https://api.mgmt.cloud.vmware.com.
- United Kingdom:  https://uk.api.mgmt.cloud.vmware.com
- Japan: https://jp.api.mgmt.cloud.vmware.com
- Singapore: https://sg.api.mgmt.cloud.vmware.com
- Germany: https://de.api.mgmt.cloud.vmware.com
- Australia: https://au.api.mgmt.cloud.vmware.com
- Canada: https://ca.api.mgmt.cloud.vmware.com
- Brazil: https://br.api.mgmt.cloud.vmware.com